backend/api/models/Badge.py

```diff
@@ -25,63 +24,200 @@ def serialize_to_json(self):
             "description": self.description,
             "congrats_text": self.congrats_text,
             "graphic": self.graphic,
+            "year": 2018,  # if we get a table in to track current year we need to use that
         }
 
 
-class FirstQuizBadge(Badge):
-    def __init__(self):
+class BadgeCategory:
+    def __init__(self, badges, name):
+        self.badges = badges
+        self.name = name
+
+    # returns progress on the next badge in this badge category
+    def get_progress_on_next_badge(self, earned_badges_set, user, quiz_result):
+        for badge in self.badges:
+            # found the next badge to earn
+            if badge.id not in earned_badges_set:
+                return (self.get_progress(user, quiz_result), badge)
+        return None
+
+    def get_newly_earned_badges(self, earned_badges_set, user, quiz_result):
+        # fill out the progress field in user
+        user.progress = self.get_progress(user, quiz_result)
+        # track earned badges
+        earned_badges = []
+        for badge in self.badges:
+            if badge.id not in earned_badges_set:
+                if badge.can_receive(user, quiz_result):
+                    earned_badges.append(badge)
+        return earned_badges
+
+    def get_progress(self, user, quiz_result):
+        return 0
+
+
+def num_quiz_attempts(user):
+    quiz_set = set()
+    for quiz_result in user.attempted_quizzes:
```

Comment:
why can't you return len(user.attempted_questions)? Will there be duplicate items in attempted quizzes?